sentry/src/main/java/io/sentry/SpotlightIntegration.java

```diff
@@ -0,0 +1,140 @@
+package io.sentry;
+
+import static io.sentry.SentryLevel.DEBUG;
+import static io.sentry.SentryLevel.ERROR;
+
+import io.sentry.util.Platform;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.zip.GZIPOutputStream;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+@ApiStatus.Internal
+public final class SpotlightIntegration
+    implements Integration, SentryOptions.BeforeEnvelopeCallback, Closeable {
+
+  private @NotNull SentryOptions options = SentryOptions.empty();
```

Comment:
`m`: shall we make it nullable rather, to not hit https://github.com/getsentry/sentry-java/issues/2541?